manager/scheduler/filter.go

```diff
@@ -61,13 +61,34 @@ func (f *ResourceFilter) SetTask(t *api.Task) bool {
 	if r == nil || r.Reservations == nil {
 		return false
 	}
-	if r.Reservations.NanoCPUs == 0 && r.Reservations.MemoryBytes == 0 {
+
+	res := r.Reservations
+	if res.NanoCPUs == 0 && res.MemoryBytes == 0 && len(res.ThirdParty) == 0 {
 		return false
 	}
+
 	f.reservations = r.Reservations
 	return true
 }
 
+// HasEnoughThirdPartyResources returns true if node can satisfy task TPR's request
+func HasEnoughThirdPartyResources(nodeRes *api.ThirdPartyResource, taskRes *api.ThirdPartyResource) bool {
+	t := taskRes.Resource.(*api.ThirdPartyResource_Integer).Integer.Val
```

Comment:
In the filter case, if there is an error (function check in the PR) should there be some kind of error handling / returning when calling this function ?
If so, how do I send this up the call stack ? It doesn't seem right to change the filter interface :)